node/src/reactor/main_reactor/catch_up.rs

```diff
@@ -120,26 +120,37 @@ impl MainReactor {
                 // if too much time has passed, the node will shutdown and require a
                 // trusted block hash to be provided via the config file
                 info!("CatchUp: local tip detected, no trusted hash");
-                if block.header().is_switch_block() {
-                    self.switch_block_header = Some(block.header().clone());
-                }
                 Either::Left(SyncIdentifier::LocalTip(
                     *block.hash(),
                     block.height(),
                     block.header().era_id(),
                 ))
             }
-            Ok(None) if self.switch_block_header.is_none() => {
-                // no trusted hash, no local block, might be genesis
-                self.catch_up_check_genesis()
-            }
             Ok(None) => {
-                // no trusted hash, no local block, no error, must be waiting for genesis
-                info!("CatchUp: waiting to store genesis immediate switch block");
-                Either::Right(CatchUpInstruction::CheckLater(
-                    "waiting for genesis immediate switch block to be stored".to_string(),
-                    self.control_logic_default_delay.into(),
-                ))
+                match self
+                    .storage
+                    .read_highest_switch_block_headers(1)
+                    .map(|headers| headers.get(0).cloned())
+                {
+                    Ok(Some(_)) => {
+                        // no trusted hash, no local block, no error, must be waiting for genesis
+                        info!("CatchUp: waiting to store genesis immediate switch block");
+                        Either::Right(CatchUpInstruction::CheckLater(
+                            "waiting for genesis immediate switch block to be stored".to_string(),
+                            self.control_logic_default_delay.into(),
+                        ))
+                    }
+                    Ok(None) => {
+                        // no trusted hash, no local block, might be genesis
+                        self.catch_up_check_genesis()
+                    }
+                    Err(storage_err) => {
+                        return Either::Right(CatchUpInstruction::Fatal(format!(
+                            "Could not read storage to find highest switch block header: {}",
```

Comment:
Nit, for the sake of consistency.

```suggestion
                            "CatchUp: could not read storage to find highest switch block header: {}",
```

Reply:
Fixed in https://github.com/casper-network/casper-node/pull/3971/commits/2b4c9ae04b1ad1c97b873fabb58367cc72e32144